src/DynamoCoreWpf/ViewModels/Core/DynamoViewModel.cs

```diff
@@ -754,6 +756,32 @@ protected DynamoViewModel(StartConfiguration startConfiguration)
             FileTrustViewModel = new FileTrustWarningViewModel();
         }
 
+        private void CurrentDispatcher_UnhandledException(object sender, DispatcherUnhandledExceptionEventArgs e)
+        {
+            if (e.Handled)
+            {
+                return;
+            }
+
+            e.Handled = true;
+            CrashGracefully(e.Exception);
+        }
+
+        private void CrashGracefully(Exception ex)
+        {
+            try
+            {
+                Model?.Logger?.LogError($"Unhandled exception {ex.Message}");
+
+                DynamoModel.IsCrashing = true;
+                Analytics.TrackException(ex, true);
+                CrashReportTool.ShowCrashErrorReportWindow(this, new Dynamo.Core.CrashErrorReportArgs(ex));
```

Comment:
we should use the crash API that has the old Dynamo crash UI as a backup  (useful when CER is not installed on the user system)

Reply:
Not sure if we should expose the CER tool API. Maybe we should, if the integrators code (not Dynamo internal) crashes maybe it is useful for integrators to generate CER report with the DYnamo API.